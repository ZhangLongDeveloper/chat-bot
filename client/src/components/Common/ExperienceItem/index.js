import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import { getDuration } from '../../../utils/common';

const ExperienceItem = ({
  image, companyName, companyLink,
  position, start, finish,
  location, description, responsibilities, tools,
}) => {
  const renderResponsibilities = () => {
    if (responsibilities && responsibilities.length) {
      return (
        <div className="responsibilities">
          <span>Main responsibilities:</span>
          <ul>
            {responsibilities.map(item => <li key={item}>{item}</li>)}
          </ul>
        </div>
      );
    }
    return null;
  };

  const renderTools = () => {
    if (tools && tools.length) {
      return (
        <div className="tools">
          <span>Technologies: </span>{tools.join(' | ')}
        </div>
      );
    }
    return null;
  };

  const renderDates = () => {
    const startDate = moment(start).format('MMM YYYY');
    const finishDate = finish ? moment(finish).format('MMM YYYY') : 'Now';
    const duration = getDuration(start, finish);

    return (
      <div className="dates-wrapper">
        <span className="dates">{startDate} - {finishDate}</span>
        <span className="duration">({duration})</span>
      </div>
    );
  };

  return (
    <div className="hobby-container row valign-wrapper">
      <div className="col s12 m3 visual">
        <img src={image} alt="icon" className="image" />
      </div>
      <div className="col s12 m9">
        <div className="title">
          <span className="position">{position}</span>
          <span className="connector">in</span>
          <a href={companyLink} className="company">{companyName}</a>
        </div>
        { renderDates() }
        <div className="location">{location}</div>
        <p className="description">{description}</p>
        { renderResponsibilities() }
        { renderTools() }
      </div>
    </div>
  );
};

ExperienceItem.propTypes = {
  image: PropTypes.string.isRequired,
  companyName: PropTypes.string.isRequired,
  companyLink: PropTypes.string.isRequired,
  position: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  start: PropTypes.object.isRequired,
  finish: PropTypes.object,
  tools: PropTypes.array,
  responsibilities: PropTypes.array,
};

ExperienceItem.defaultProps = {
  finish: undefined,
  tools: null,
  responsibilities: null,
};

export default ExperienceItem;
