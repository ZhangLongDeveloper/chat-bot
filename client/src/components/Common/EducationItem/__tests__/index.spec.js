import React from 'react';
import { create } from 'react-test-renderer';

import ExperienceItem from '../../ExperienceItem';

jest.mock('../../../../utils/common', () => ({
  getDuration: jest.fn(() => 'getDuration mocked result'),
}));

const mockComponent = props => (
  <ExperienceItem {...props} />
);

describe('components/Common/<ExperienceItem />', () => {
  it('should render component with basic data', () => {
    const props = {
      image: 'image',
      companyName: 'companyName',
      companyLink: 'companyLink',
      position: 'position',
      location: 'location',
      description: 'description',
      start: 'start',
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with all the data', () => {
    const props = {
      image: 'image',
      companyName: 'companyName',
      companyLink: 'companyLink',
      position: 'position',
      location: 'location',
      description: 'description',
      start: 'start',
      finish: 'finish',
      tools: ['tools1', 'tools2'],
      responsibilities: ['responsibilities1', 'responsibilities2', 'responsibilities3'],
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
