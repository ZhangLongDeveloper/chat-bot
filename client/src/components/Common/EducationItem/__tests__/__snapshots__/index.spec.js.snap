// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Common/<ExperienceItem /> should render component with all the data 1`] = `
<div
  className="experience-container row valign-wrapper"
>
  <div
    className="col s12 m3 visual"
  >
    <a
      href="companyLink"
    >
      <img
        alt="icon"
        className="image"
        src="image"
      />
    </a>
  </div>
  <div
    className="col s12 m9"
  >
    <div
      className="title"
    >
      <span
        className="position"
      >
        position
      </span>
      <span
        className="connector"
      >
        in
      </span>
      <a
        className="company"
        href="companyLink"
      >
        companyName
      </a>
    </div>
    <div
      className="dates-wrapper"
    >
      <span
        className="dates"
      >
        Invalid date
         - 
        Invalid date
      </span>
      <span
        className="duration"
      >
        (
        getDuration mocked result
        )
      </span>
    </div>
    <div
      className="location"
    >
      location
    </div>
    <p
      className="description"
    >
      description
    </p>
    <div
      className="responsibilities"
    >
      <div
        className="responsibilities-label"
      >
        Main responsibilities:
      </div>
      <ul>
        <li>
          responsibilities1
        </li>
        <li>
          responsibilities2
        </li>
        <li>
          responsibilities3
        </li>
      </ul>
    </div>
    <div
      className="tools"
    >
      <div
        className="tools-label"
      >
        Technologies: 
      </div>
      <div>
        tools1 | tools2
      </div>
    </div>
  </div>
</div>
`;

exports[`components/Common/<ExperienceItem /> should render component with basic data 1`] = `
<div
  className="experience-container row valign-wrapper"
>
  <div
    className="col s12 m3 visual"
  >
    <a
      href="companyLink"
    >
      <img
        alt="icon"
        className="image"
        src="image"
      />
    </a>
  </div>
  <div
    className="col s12 m9"
  >
    <div
      className="title"
    >
      <span
        className="position"
      >
        position
      </span>
      <span
        className="connector"
      >
        in
      </span>
      <a
        className="company"
        href="companyLink"
      >
        companyName
      </a>
    </div>
    <div
      className="dates-wrapper"
    >
      <span
        className="dates"
      >
        Invalid date
         - 
        Now
      </span>
      <span
        className="duration"
      >
        (
        getDuration mocked result
        )
      </span>
    </div>
    <div
      className="location"
    >
      location
    </div>
    <p
      className="description"
    >
      description
    </p>
  </div>
</div>
`;
